### DISCUSSION TITLE
blog/onion-architecture-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#35] [General]
# blog/onion-architecture-in-aspnet-core/

In this article, We will talk about Onion Architecture In ASP.NET Core and its advantages. We will also together build a WebApi that follows a variant of Onion Architecture so that we get to see why it is important to implement such an architecture in your upcoming projects.

https://codewithmukesh.com/blog/onion-architecture-in-aspnet-core/

**Comment by Yousha** [comment]
Tier != Layer

They are NOT the same.
Layer is functional/logical division of a software on SAME infrastructure. Like Ms Office, a multi-layered but single-tire software.
Tier is physical division of a software on DIFFERENT infrastructures. Like:
Put logic tier in machine #1
The presentation tier in machine #2
And the data tier in machine #3

**Comment by YashSmithh** [comment]
Onion Architecture in ASP.NET Core with CQRS separates concerns into layers—Core, Infrastructure, and Application—enhancing maintainability and testability. CQRS divides read and write operations, improving performance and scalability. Partner with dev technosys or <a href="https://devtechnosys.com/hire-developers-in-ireland.php"> hire developers in ireland</a> to build robust, scalable ASP.NET Core solutions using these modern architectural patterns.
